src/libraries/System.Runtime/tests/System/TimeOnlyTests.cs

```diff
@@ -0,0 +1,478 @@
+            Assert.Equal(10, to.Minute);
+            Assert.Equal(30, to.Second);
+
+            to = to.AddMinutes(-4 * 60, out wrappedDays);
+            Assert.Equal(23, to.Hour);
+            Assert.Equal(10, to.Minute);
+            Assert.Equal(30, to.Second);
+            Assert.Equal(-1, wrappedDays);
+
+            to = to.AddMinutes(60.5, out wrappedDays);
+            Assert.Equal(0, to.Hour);
+            Assert.Equal(11, to.Minute);
+            Assert.Equal(0, to.Second);
+            Assert.Equal(1, wrappedDays);
+        }
+
+        [Fact]
+        public static void IsBetweenTest()
+        {
+            TimeOnly to1 = new TimeOnly(14, 30);
+            TimeOnly to2 = new TimeOnly(2, 0);
+            TimeOnly to3 = new TimeOnly(12, 0);
+
+            Assert.True(to3.IsBetween(to2, to1));
+            Assert.True(to1.IsBetween(to3, to2));
+
+            Assert.True(to3.IsBetween(to3, to1));
+            Assert.True(to1.IsBetween(to1, to2));
+
+            Assert.False(to1.IsBetween(to3, to1));
+            Assert.False(to2.IsBetween(to3, to2));
+
+            Assert.True(to1.IsBetween(to3, to1.Add(new TimeSpan(1))));
+            Assert.True(to2.IsBetween(to3, to2.Add(new TimeSpan(1))));
+        }
+
+        [Fact]
+        public static void CompareOperatosTest()
+        {
+            TimeOnly to1 = new TimeOnly(14, 30);
+            TimeOnly to2 = new TimeOnly(14, 30);
+            TimeOnly to3 = new TimeOnly(14, 30, 1);
+
+            Assert.True(to1 == to2);
+            Assert.True(to1 >= to2);
+            Assert.True(to1 <= to2);
+
+            Assert.True(to1 != to3);
+            Assert.True(to1 < to3);
+            Assert.True(to1 <= to3);
+
+            Assert.True(to3 > to1);
+            Assert.True(to3 >= to1);
+
+            Assert.False(to1 == to3);
+            Assert.False(to1 > to3);
+            Assert.False(to3 < to1);
+            Assert.False(to1 != to2);
+        }
+
+        [Fact]
+        public static void SubtractOperatorTest()
+        {
+            TimeOnly to1 = new TimeOnly(10, 30, 40);
+            TimeOnly to2 = new TimeOnly(14, 0);
+
+            Assert.Equal(new TimeSpan(3, 29, 20), to1 - to2);
+            Assert.Equal(new TimeSpan(3, 29, 20), to2 - to1);
+            Assert.Equal(TimeSpan.Zero, to1 - to1);
+        }
+
+        [Fact]
+        public static void FromToTimeSpanTest()
+        {
+            Assert.Equal(TimeOnly.MinValue, TimeOnly.FromTimeSpan(TimeSpan.Zero));
+            Assert.Equal(TimeSpan.Zero, TimeOnly.MinValue.ToTimeSpan());
+
+            Assert.Equal(new TimeOnly(10, 20, 30), TimeOnly.FromTimeSpan(new TimeSpan(10, 20, 30)));
+            Assert.Equal(new TimeSpan(14, 10, 50), new TimeOnly(14, 10, 50).ToTimeSpan());
+
+            Assert.Equal(TimeOnly.MaxValue, TimeOnly.FromTimeSpan(TimeOnly.MaxValue.ToTimeSpan()));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => TimeOnly.FromTimeSpan(new TimeSpan(24, 0, 0)));
```

Comment:
Nit, up up you, but I believe there is an AssertExtensions.Throws that validates the parameter in the ArgumentException